src/utils/components/HeadlessMode/HeadlessMode.tsx

```diff
@@ -10,10 +10,7 @@ type HeadlessModeProps = {
 
 const HeadlessMode: FC<HeadlessModeProps> = ({ updateHeadlessMode, vm }) => {
   const devices = vm?.spec?.template?.spec?.domain?.devices;
-  const [isChecked, setIsChecked] = useState<boolean>(
-    devices?.hasOwnProperty('autoattachGraphicsDevice') &&
-      devices?.autoattachGraphicsDevice === false,
-  );
+  const [isChecked, setIsChecked] = useState<boolean>(devices?.autoattachGraphicsDevice === false);
```

Comment:
same here if isHeadlessModeVMI(vm?.spec?.template) works the same please change to utilize the util function :)